src/main/java/engineering/swat/watch/Watcher.java

```diff
@@ -188,4 +199,29 @@ public ActiveWatch start() throws IOException {
                 throw new IllegalStateException("Not supported yet");
         }
     }
+
+    private BiConsumer<EventHandlingWatch, WatchEvent> overflowEventHandler() {
+        switch (overflowPolicy) {
+            case NO_RESCANS:
+                return (w, e) -> {};
+            case MEMORYLESS_RESCANS:
+                return new MemorylessRescanner(executor);
+            default:
+                throw new UnsupportedOperationException("No event handler has been defined yet for this overflow policy");
+        }
+    }
+
+    private BiConsumer<EventHandlingWatch, WatchEvent> userDefinedEventHandler() {
+        // If overflow events are auto-handled because of the overflow policy,
```

Comment:
I think you should always get an overflow event, no matter which heuristic we use to help you deal with it.

It's up to a user to ignore the overflow event, not up to us.